robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -278,6 +280,30 @@ def distance(e1, e2):
         return e1.distance_to_3d(e2._pose.p)
 
 
+class NavigateToPreGrasp(NavigateToDesignator):
+    """Navigate so that the arm can reach the grasp entity and we can see the entity
+    :param robot: robot object
+    :param entity_designator: designator that resolves to an Ed Entity
+    :param arm_designator: which arm to eventually grasp with?
+    """
+    def __init__(self, robot, entity_designator, arm_designator=None):
+        constraint_designator = CompoundConstraintsDesignator()
+        constraint_designator.add(ArmsreachConstraintsDesignator(robot, entity_designator, arm_designator, look=False), 'reach')
+        constraint_designator.add(LookAtConstraintsDesignator(entity_designator, 1.57), 'face') #TODO hero-specific magic number
+        super(NavigateToPreGrasp, self).__init__(robot, constraint_designator, reset_head=True)
+
+
+class NavigateToGrasp(NavigateToDesignator):
```

Comment:
Making it slightly less verbose is worth it in my opinion since several challenges make use of it. And while they may not be very verbose now that might change if we plan to utilise more constraints, e.g. the fact that the robot needs line-of-sight to the object.

You are correct on the 'reset_head' parameter, I will expose that